user/tests/wiring/ble/ibeacon/application.cpp

```diff
@@ -0,0 +1,38 @@
+/*
+ ******************************************************************************
+  Copyright (c) 2015 Particle Industries, Inc.  All rights reserved.
+
+  This program is free software; you can redistribute it and/or
+  modify it under the terms of the GNU Lesser General Public
+  License as published by the Free Software Foundation, either
+  version 3 of the License, or (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+  Lesser General Public License for more details.
+
+  You should have received a copy of the GNU Lesser General Public
+  License along with this program; if not, see <http://www.gnu.org/licenses/>.
+  ******************************************************************************
+ */
+
+/* Includes ------------------------------------------------------------------*/
+#include "application.h"
+
+uint8_t beaconUuid[16] = {0x00,0x11,0x22,0x33,0x44,0x55,0x66,0x77,0x88,0x99,0xaa,0xbb,0xcc,0xdd,0xee,0xff};
+
+void setup() {
+    iBeacon beacon(1, 2, beaconUuid, -30);
+    BLE.advertisementData(beacon);

```

Comment:
This should work more like the heartrate peripheral — set up the iBeacon, no need for `advertisementData`, `advertise()` just does the right thing.

Reply:
The advertising data is not just an array. It must follow a particular pattern so that the BLE stack will adopt it. Here are the APIs to let user the advertising data safely: https://github.com/particle-iot/device-os/pull/1728/files?#diff-75b7ed256204b5221da93d699fee0429R219